packages/rocketchat-cas/cas_server.js

```diff
@@ -8,80 +11,80 @@ RoutePolicy.declare('/_cas/', 'network');
-    Fiber(function () {
-        middleware(req, res, next);
-    }).run();
+	// Need to create a Fiber since we're using synchronous http calls and nothing
+	// else is wrapping this in a fiber automatically
+	Fiber(function () {
+		middleware(req, res, next);
+	}).run();
 });
 
 var middleware = function (req, res, next) {
-    // Make sure to catch any exceptions because otherwise we'd crash
-    // the runner
-    try {
-        var barePath = req.url.substring(0, req.url.indexOf('?'));
-        var splitPath = barePath.split('/');
-
-        // Any non-cas request will continue down the default
-        // middlewares.
-        if (splitPath[1] !== '_cas') {
-            next();
-            return;
-        }
-
-        // get auth token
-        var credentialToken = splitPath[2];
-        if (!credentialToken) {
-            closePopup(res);
-            return;
-        }
-
-        // validate ticket
-        casTicket(req, credentialToken, function() {
-            closePopup(res);
-        });
-
-    } catch (err) {
-        logger.error("Unexpected error : " + err.message);
-        closePopup(res);
-    }
+	// Make sure to catch any exceptions because otherwise we'd crash
+	// the runner
+	try {
+		var barePath = req.url.substring(0, req.url.indexOf('?'));
+		var splitPath = barePath.split('/');
+
+		// Any non-cas request will continue down the default
+		// middlewares.
+		if (splitPath[1] !== '_cas') {
+			next();
+			return;
+		}
+
+		// get auth token
+		var credentialToken = splitPath[2];
+		if (!credentialToken) {
+			closePopup(res);
+			return;
+		}
+
+		// validate ticket
+		casTicket(req, credentialToken, function() {
+			closePopup(res);
+		});
+
+	} catch (err) {
+		logger.error('Unexpected error : ' + err.message);
+		closePopup(res);
+	}
 };
 
 var casTicket = function (req, token, callback) {
 
-    // get configuration
-    if (!RocketChat.settings.get("CAS_enabled")) {
-        logger.error("Got ticket validation request, but CAS is not enabled");
-        callback();
-    }
-
-    // get ticket and validate.
-    var parsedUrl = url.parse(req.url, true);
-    var ticketId = parsedUrl.query.ticket;
-    var baseUrl = RocketChat.settings.get("CAS_base_url");
-    logger.debug("Using CAS_base_url: " + baseUrl);
-
-    var cas = new CAS({
-        base_url: baseUrl,
-        service: Meteor.absoluteUrl() + "_cas/" + token
-    });
-
-    cas.validate(ticketId, function(err, status, username) {
-        if (err) {
-            logger.error("error when trying to validate " + err);
-        } else {
-            if (status) {
-                logger.info("Validated user: " + username);
-                _casCredentialTokens[token] = { id: username };
-            } else {
-                logger.error("Unable to validate ticket: " + ticketId);
-            }
-        }
-
-        callback();
-  });
-
-  return; 
+	// get configuration
+	if (!RocketChat.settings.get('CAS_enabled')) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["RocketChat" is not defined.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2313371398/issues/source?bid=3199634&fileBranchId=3199636#l56)